year-2022/day-03/src/main.rs

```diff
@@ -0,0 +1,133 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    const TEST_INPUT: &str = "vJrwpWtwJgWrhcsFMMfFFhFp
+jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL
+PmmdzqPrVvPwwTWBwg
+wMqvLMZHhHMvwLHjbvcjnnSBnvTQFn
+ttgJtRGJQctTZtZT
+CrZsJsPPZsGzwwsLwLmpwMDw";
+
+    #[test]
+    fn test_solution() {
+        let backpacks = parse_content(TEST_INPUT);
+        let compartments = split_into_compartments(&backpacks);
+
+        let duplicates_sum = sum_duplicates_priority(&compartments);
+        assert_eq!(duplicates_sum, 157);
+
+        let groups = split_into_groups(&backpacks);
+        let badges_sum = sum_badges_priority(&groups);
+        assert_eq!(badges_sum, 70);
+    }
+}
+
+const GROUP_SIZE: usize = 3;
+
+fn parse_content(content: &str) -> Vec<&str> {
+    let mut backpacks = vec![];
+
+    for items in content.split('\n') {
+        backpacks.push(items);
+    }
+
+    return backpacks;
+}
+
+fn split_into_compartments<'a>(backpacks: &'a [&str]) -> Vec<(&'a str, &'a str)> {
+    let mut compartments = vec![];
+
+    for items in backpacks {
+        let index = items.len() / 2;
+
+        let left = &items[..index];
+        let right = &items[index..];
+
+        compartments.push((left, right));
+    }
+
+    return compartments;
+}
+fn split_into_groups<'a>(backpacks: &'a [&str]) -> Vec<Vec<&'a str>> {
+    let mut groups = vec![];
+
+    for chunk in backpacks.chunks(GROUP_SIZE) {
+        groups.push(chunk.to_vec());
+    }
+
+    return groups;
+}
+
+fn find_duplicate_item(backpack: (&str, &str)) -> char {
```

Comment:
Come mai hai tolto l'`Option<T>` ?

Reply:
Ho usato il metodo `unwrap` che ho compreso in maniera decente con la soluzione del 4° giorno; dando per assodato che -appunto- avrei sempre trovato una soluzione, mi sembrava superfluo usare l'`Option`...

---

Il messaggio dell'ultimo commit, infatti, voleva essere:
> It makes so much more sense with the use of the `unwrap` function. 😅

... ma ho dimenticato l'escape dei caratteri `backtick` e Bash ha cercato di interpretarlo come comando; risultando in una stringa vuota! 😅

Reply:
ah ecco!